packages/typespec-python/generator/pygen/codegen/serializers/__init__.py

```diff
@@ -555,6 +555,10 @@ def _serialize_and_write_test(self, env: Environment, namespace_path: Path):
 
         for client in self.code_model.clients:
             for og in client.operation_groups:
+                if self.code_model.options["multiapi"] and any(
+                    o.api_versions[0] != self.code_model.options["default_api_version"] for o in og.operations
+                ):
```

Comment:
only need to generate test for default api-version like we do for generate sample